sdk/core/azure-core/azure/core/pipeline/policies/_retry_async.py

```diff
@@ -156,6 +158,9 @@ async def send(self, request):  # pylint:disable=invalid-overridden-method
                 # the authentication policy failed such that the client's request can't
                 # succeed--we'll never have a response to it, so propagate the exception
                 raise
+            except (ServiceRequestTimeoutError, ServiceResponseTimeoutError):
```

Comment:
Given this scenario, timeout setting is 10 min and default timeout is 5 min.

We will fail on the second TimeoutError but not the first one, right?

Reply:
I don't understand what you mean by a first vs. second error. `_configure_timeout` raises one of these errors when `absolute_timeout <= 0`. I believe we never want to retry when that's true.

Reply:
I meant (ServiceRequestTimeoutError, ServiceResponseTimeoutError) might be raised from transport. In this case, absolute_timeout may still >0 and we do want to retry.

Reply:
Nothing else in the library raises these errors today but it's a fair point that a transport or policy could do so. I've removed this except block in favor of checking `absolute_timeout` in the broader handler.